src/app/qgsmaptoolrotatefeature.cpp

```diff
@@ -139,24 +141,30 @@ void QgsAngleMagnetWidget::angleSpinBoxValueChanged( double angle )
 //
 
 QgsMapToolRotateFeature::QgsMapToolRotateFeature( QgsMapCanvas *canvas )
-  : QgsMapToolEdit( canvas )
+  : QgsMapToolAdvancedDigitizing( canvas, QgisApp::instance()->cadDockWidget() )
+  , mSnapIndicator( std::make_unique< QgsSnapIndicator>( canvas ) )
 {
+  mToolName = tr( "Rotate feature" );
 }
 
 QgsMapToolRotateFeature::~QgsMapToolRotateFeature()
 {
   deleteRotationWidget();
   mAnchorPoint.reset();
   deleteRubberband();
+  mSnapIndicator->setMatch( QgsPointLocator::Match() );
 }
 
-void QgsMapToolRotateFeature::canvasMoveEvent( QgsMapMouseEvent *e )
+void QgsMapToolRotateFeature::cadCanvasMoveEvent( QgsMapMouseEvent *e )
 {
+  mSnapIndicator->setMatch( e->mapPointMatch() );
+
   if ( mRotationActive )
   {
-    const double XDistance = e->pos().x() - mStPoint.x();
-    const double YDistance = e->pos().y() - mStPoint.y();
-    double rotation = std::atan2( YDistance, XDistance ) * ( 180 / M_PI ) - mRotationOffset;
+    const QgsPointXY pt = e->mapPoint();
+    const double XDistance = pt.x() - mStartPointMapCoords.x();
+    const double YDistance = pt.y() - mStartPointMapCoords.y();
+    double rotation = std::atan2( XDistance, YDistance ) * ( 180 / M_PI ) - mRotationOffset;
```

Comment:
Why the X/y flip?

Reply:
I'm not exactly sure, didn't investigate it. Switching from `e->pos()` screen coordinates to `e->mapPoint()` map coordinates reversed the rotation direction, so I flipped the atan.

Reply:
Maybe a result a moving from y top to bottom painter coordinates to y bottom to top geographic coordinates?

Reply:
That was what I suspected but wasn't sure if painter coords y axis was inverted.

Reply:
Is there a crs with an inverted y axis? Maybe I should test it.

Reply:
There are some south african ones, but QGIS doesn't support these well in general. See https://github.com/qgis/QGIS/issues/32391